docs/getstarted/upgrade-to-upbound/migrate-configurations-v2.md

```diff
@@ -0,0 +1,804 @@
+---
+title: Migrate Crossplane v1 Configurations to v2
+description: Migrate XRDs, compositions, and KCL functions from Crossplane v1 to v2 with namespaced resources
+sidebar_position: 4
+---
+
+This guide documents the migration from Crossplane v1 to v2 for configurations,
+including all breaking changes and lessons learned. It covers updating XRDs,
+compositions, KCL functions, and tests for the new v2 namespaced resource model.
+
+## Overview
+
+Crossplane v2 introduces **namespaced resources** with the `.m` API group suffix
+(for example, `rds.aws.m.upbound.io/v1beta1`) and removes the XR/Claim separation
+pattern. This migration covers updating XRDs, compositions, KCL functions, and
+tests.
+
+## Prerequisites
+
+- Upgrade `upbound.yaml` to reference v2 provider packages
+- Run `up project build` to generate v2 models in `.up/kcl/models/`
+- Ensure Docker is running for composition rendering tests
+
+## Breaking changes
+
+### 1. XRD API version and structure
+
+**Before (v1):**
+```yaml
+apiVersion: apiextensions.crossplane.io/v1
+kind: CompositeResourceDefinition
+metadata:
+  name: xsqlinstances.aws.platform.upbound.io
+spec:
+  claimNames:
+    kind: SQLInstance
+    plural: sqlinstances
+  connectionSecretKeys:
+    - username
+    - password
+  group: aws.platform.upbound.io
+  names:
+    kind: XSQLInstance
+    plural: xsqlinstances
+```
+
+**After (v2):**
+```yaml
+apiVersion: apiextensions.crossplane.io/v2
+kind: CompositeResourceDefinition
+metadata:
+  name: sqlinstances.aws.platform.upbound.io
+spec:
+  scope: Namespaced  # NEW: Required in v2
+  group: aws.platform.upbound.io
+  names:
+    kind: SQLInstance  # Removed "X" prefix
+    plural: sqlinstances
+  # claimNames removed - claims not supported in v2
+  # connectionSecretKeys moved or handled differently
```

Comment:
can we link here somehow https://github.com/crossplane/docs/pull/1034 after it gets merged ?!